ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh

```diff
@@ -0,0 +1,451 @@
+function get_new_caps() {
+    # echo "Getting new caps introduced in target release as compared to source release..."
+    local source="$1"
+    local target="$2"
+    local array_source array_target extra_caps
+
+    # Check for missing environment variables
+    if [ -z "$source" ] || [ -z "$target" ]; then
+        echo "ERROR: Missing required arguments"
+        return 1
+    fi
+
+    # Split the elements into arrays
+    IFS=" " read -ra array_source <<< "${source}"
+    IFS=" " read -ra array_target <<< "${target}"
+
+    # Find the extra elements in $target compared to $source
+    extra_caps=()
+    for cap in "${array_target[@]}"; do
+        #shellcheck disable=SC2076
+        if [[ ! " ${array_source[*]} " =~ " ${cap} " ]]; then
+            extra_caps+=("$cap")
+        fi
+    done
+
+    # Combine the extra elements into a space-separated string
+    echo "${extra_caps[*]}"
+    #echo "Expected implicitly enabled capabilities list is ${expected_implicit_caps}"
+}
+
+# compares two payloads for capabilities difference at manifest level
+function extract_migrated_caps() {
+    echo "Extracting migrated caps between 2 releases..."
+    local source="$1"
+    local target="$2"
+    local source_dir target_dir source_cap_files
+
+    # Check for missing environment variables
+    if [ -z "$source" ] || [ -z "$target" ]; then
+        echo "ERROR: Missing required arguments"
+        return 1
+    fi
+
+    source_dir="/tmp/source"
+    target_dir="/tmp/target"
+    mkdir -p ${source_dir} ${target_dir}
+
+    if ! extract_manifests "${source_dir}" "${source}" || ! extract_manifests "${target_dir}" "${target}"; then
+        return 1
+    fi
+
+    source_cap_files=$(grep -rl "capability.openshift.io/name" ${source_dir})
+    for source_file in ${source_cap_files}; do
+        filename=$(basename "${source_file}")
+        target_file="$target_dir/$filename"
+
+        if [ -f "$target_file" ]; then
+            source_cap=$(grep 'capability.openshift.io/name:' "$source_file" | cut -d ':' -f2 | tr -d '"' | sort | uniq )
+            target_cap=$(grep 'capability.openshift.io/name:' "$target_file" | cut -d ':' -f2 | tr -d '"' | sort | uniq )
+
+            if [ "$source_cap" != "$target_cap" ]; then
+                echo "Migrated capability in $filename: $source_cap -> $target_cap"
+                migrated_caps["$source_cap"]=$target_cap
+            fi
+        fi
+    done
+}
+
+function check_cvo_cap() {
+
+    local result=0
+    local capability_set=$1
+    local expected_status=$2
+    local cvo_field=$3
+
+    cvo_caps=$(oc get clusterversion version -o json | jq -rc "${cvo_field}")
+    if [[ "${capability_set}" == "" ]] && [[ "${cvo_caps}" != "null" ]]; then
+        echo "ERROR: ${expected_status} capability set is empty, but find capabilities ${cvo_caps} in cvo ${cvo_field}"
+        result=1
+    fi
+    if [[ "${capability_set}" != "" ]]; then
+        if [[ "${cvo_caps}" == "null" ]]; then
```

Comment:
Can we extract the if condition to a separate block just like what we did on line 212 for easy reading?

Reply:
Okay, ignore it. I saw `cucushift-installer-check-capability` is also using the same syntax, if needed, we can update them together later.